lib/miq_automation_engine/engine/miq_ae_event.rb

```diff
@@ -81,6 +81,13 @@ def self.build_evm_event(event, passed_inputs = {})
       aevent.merge!("#{hash[:class].name}::#{hash[:name]}" => vmdb_object.id, "#{hash[:key]}_id".to_sym  => vmdb_object.id)
     end
 
+    inputs.each do |k, value|
+      next unless value.kind_of?(ApplicationRecord)
+      obj = inputs.delete(k)
```

Comment:
@lfu You do not want to call delete on the object you are enumerating over.  This will cause it to skip over objects in `inputs`.  You should use `delete_if` instead.